### PR TITLE
Improve order processing

### PR DESCRIPTION
**Pull request by emtoptal**
- Use EventBridge for Shopify events propagation and send an alarm to the SNS topic if the AuthBot lambda function shows more than 0 errors over the last 5 minutes.

- Tags with the «authbot:» prefix will be treated as group names.  If the group doesn't exist, it will be automatically created.

### CODE REVIEW

File: terraform/templates/eventbridge/main.tf [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#
```
Comment (hndrewaall) [inline terraform/templates/eventbridge/main.tf:1]: much nicer with EventBridge! thanks for doing this

File: terraform/templates/retrobot/main.tf [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    secrets = {
+      actions   = ["secretsmanager:GetSecretValue"]
+      effect    = "Allow"
+      resources = [var.shopify_pass_arn]
+    }
+  }
+}
+
+# module "topic" {
```
Comment (hndrewaall) [inline terraform/templates/retrobot/main.tf:45]: what are these guys (and why are they commented out)?

File: authbot2/authbot.py [hunk 3]
```diff
@@ -58,104 +42,100 @@ def generate_temporary_password() -> str:
...
-            logger.info(f"Found product {item['product_id']}")
-            if SYNC_MAP.get(item['product_id']):
-                users[customer_email].add(SYNC_MAP[item['product_id']])
-                fulfillments[order['id']].add(item['id'])
+    order = event["detail"]["payload"]
+    order_id = order["id"]
+    logger.info(f"Processing order ID {order_id}")
+    customer_email = order.get("customer", {}).get("email")
+    if not customer_email:
+        logger.warn(f"Oder {order_id} doesn't have customer email! Exiting.")
```
Comment (hndrewaall) [inline authbot2/authbot.py]:
```suggestion
        logger.warn(f"Order {order_id} doesn't have customer email! Exiting.")
```

**Review by hndrewaall — CHANGES_REQUESTED** [review]
Beautiful. Just some minor tweaks. Excited to see it running end to end!

File: terraform/templates/retrobot/main.tf (hunk 2 above)
Comment (emtoptal) [reply terraform/templates/retrobot/main.tf:45]: This is left to be considered after the GitHub action is implemented.

File: authbot2/authbot.py (hunk 3 above)
Comment (emtoptal) [reply authbot2/authbot.py]: Good catch!

**Review by hndrewaall — DISMISSED** [review]

**Pull request closed without merging**